### PR TITLE
Fix/0.19.0 sdl

### PR DESCRIPTION
**Pull request by diptorupd**
Fixes SDL issues identified in the 0.19.0 tag.

### CODE REVIEW

**Review by mingjie-intel — CHANGES_REQUESTED** [review]

**Comment by mingjie-intel** [comment]
Ran SDL scan with this branch: 

[numba_dpex_0.19.0.pdf](https://github.com/IntelPython/numba-dpex/files/10115331/numba_dpex_0.19.0.pdf)

**Comment by mingjie-intel** [comment]
rerun SDL scan after fix: 
[numba_dpex_0.19.0 (1).pdf](https://github.com/IntelPython/numba-dpex/files/10116906/numba_dpex_0.19.0.1.pdf)

**Comment by mingjie-intel** [comment]
[numba_dpex_0.19.0 (3).pdf](https://github.com/IntelPython/numba-dpex/files/10189401/numba_dpex_0.19.0.3.pdf)

rerun SDL scan. Here are what can't be fixed:
1. warnings from versioneer.py.
2. allocate array using object.
3. size of pointer. 
4. print out message from system tools.

File: numba_dpex/_version.py [hunk 1]
```diff
@@ -151,21 +151,20 @@ def git_get_keywords(versionfile_abs):
...
-            if line.strip().startswith("git_full ="):
-                mo = re.search(r'=\s*"(.*)"', line)
-                if mo:
-                    keywords["full"] = mo.group(1)
-            if line.strip().startswith("git_date ="):
-                mo = re.search(r'=\s*"(.*)"', line)
-                if mo:
-                    keywords["date"] = mo.group(1)
-        f.close()
+        with open(versionfile_abs, "r") as f:
```
Comment (diptorupd) [inline numba_dpex/_version.py:154]: This file is generated by versioneer.py and will get overwritten.

File: numba_dpex/dpctl_iface/usm_allocators_ext.c [hunk 2]
```diff
@@ -50,8 +50,6 @@ void *save_queue_allocator(size_t size, void *opaque)
     // Use that queue to allocate.
     void *data = (void *)DPCTLmalloc_shared(new_size, cur_queue);
     if (data == NULL) {
-        fprintf(stderr,
```
Comment (diptorupd) [inline numba_dpex/dpctl_iface/usm_allocators_ext.c]: does it complain if we have std::cerr?

File: numba_dpex/_version.py (hunk 1 above)
Comment (mingjie-intel) [reply numba_dpex/_version.py:154]: I see. Reverted back to original format.

File: numba_dpex/dpctl_iface/usm_allocators_ext.c (hunk 2 above)
Comment (mingjie-intel) [reply numba_dpex/dpctl_iface/usm_allocators_ext.c]: Tried something like this: 
```

fprintf(stderr, "Fatal: failed to allocate memory for AllocatorImpl.\n");
```
Checkmarx doesn't like it: Improper Resource Access Authorization.

Comment (mingjie-intel) [reply numba_dpex/dpctl_iface/usm_allocators_ext.c]: std::cerr need to include <iostream> from cpp, maybe not mix c++ and c just for this output?

**Pull request merged**